tests/unit/test_charm.py

```diff
@@ -387,3 +387,99 @@ def test_datasource(
     assert harness.charm._datasource() == expected_output
+
+
+@pytest.mark.parametrize(
+    "files, expected",
+    [
+        (
+            [
+                "/layers/paketo-buildpacks_bellsoft-liberica/jre/bin/java",
+                "/workspace/org/springframework/boot/loader/JarLauncher.class",
+                "/workspace/BOOT-INF/lib/mysql-connector-j-10.1.1.jar",
+            ],
+            True,
+        ),
+        (
+            [
+                "/layers/paketo-buildpacks_bellsoft-liberica/jre/bin/java",
+                "/workspace/org/springframework/boot/loader/JarLauncher.class",
+            ],
+            False,
+        ),
+    ],
+)
+def test_buildpack_app_mysql_capability(
+    harness: Harness, patch: SpringBootPatch, files: list[str], expected: bool
+) -> None:
+    """
+    arrange: provide a simulated OCI image mimicking a Spring Boot application image created by
+        buildpack.
+    act: start the charm.
+    assert: Spring Boot charm should finish the reconciliation process without an error.
+    """
+    app = OCIImageMock.builder()
+    for file in files:
+        app.add_file(file, b"")
+    patch.start({"spring-boot-app": app.build()})
+    harness.begin_with_initial_hooks()
+    assert harness.charm._detect_mysql_capability() == expected
+
+
+@pytest.mark.parametrize(
+    "files, expected",
+    [
+        (
+            [
+                "META-INF/",
+                "META-INF/MANIFEST.MF",
+                "org/",
+                "org/springframework/boot/loader/",
+                "org/springframework/boot/loader/ClassPathIndexFile.class",
+                "BOOT-INF/classes/com/canonical/sampleapp/",
+                "BOOT-INF/classes/com/canonical/sampleapp/Application.class",
+                "BOOT-INF/lib/",
+                "BOOT-INF/lib/log4j-to-slf4j-2.19.0.jar",
+                "BOOT-INF/lib/mysql-connector-j-2.19.0.jar",
+            ],
+            True,
+        ),
+        (
+            [
+                "META-INF/",
+                "META-INF/MANIFEST.MF",
+                "org/",
+                "org/springframework/boot/loader/",
+                "org/springframework/boot/loader/ClassPathIndexFile.class",
+                "BOOT-INF/classes/com/canonical/sampleapp/",
+                "BOOT-INF/classes/com/canonical/sampleapp/Application.class",
+                "BOOT-INF/lib/",
+                "BOOT-INF/lib/log4j-to-slf4j-2.19.0.jar",
+            ],
+            False,
+        ),
+    ],
+)
+def test_executable_jar_mysql_capability(
+    harness: Harness, patch: SpringBootPatch, files: list[str], expected: bool
+) -> None:
+    """
+    arrange: put a jar file in the /app dir of the simulated Spring Boot application container.
+    act: start the charm.
+    assert: Spring Boot charm should finish the reconciliation process without an error.
+    """
+    patch.start(
+        {"spring-boot-app": OCIImageMock.builder().add_file("/app/test.jar", b"").build()},
```

Comment:
Would it be worth putting`OCIImageMock.builder().add_file("/app/test.jar", b"").build()` on another line?

Reply:
Black puts it back on one line

Reply:
To clarify:

```python
mock_image: OCIImageMock.builder().add_file("/app/test.jar", b"").build()
patch.start(...)
```

Reply:
Ah, I understand. Fixed